api/config/v1alpha1/shared_types.go

```diff
@@ -75,6 +75,21 @@ type KubernetesPodSpec struct {
 	//
 	// +optional
 	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
+
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +optional
+	// +mapType=atomic
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
can we do without `NodeSelector` and rely on `Affinity` ?

Reply:
`corev1.PodSpec` define them as different property, so I think we should keep this.

Reply:
reframing, which use case does `NodeSelector` enable that `Affinity` cannot do today ?

Reply:
xref: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector

they are similar, but should we keep the same for the k8s API?

Reply:
I would vote to rm `nodeSelector` from this PR, since afaik the use cases can be solved with `affinity` and even the user has not requested this yet

Reply:
cc @AliceProxy @zirain @Xunzhuo